src/main/scala/com/exasol/spark/rdd/ExasolRDD.scala

```diff
@@ -48,17 +67,25 @@ class ExasolRDD(
     // Close Exasol main connection when SparkContext finishes. This is a lifetime of a Spark
     // application.
     sc.addSparkListener(new SparkListener {
-      override def onApplicationEnd(appEnd: SparkListenerApplicationEnd): Unit = conn.close()
+      override def onApplicationEnd(appEnd: SparkListenerApplicationEnd): Unit =
+        closeMainResources()
     })
 
     conn
   }
 
+  @SuppressWarnings(Array("org.wartremover.warts.AsInstanceOf"))
   override def getPartitions: Array[Partition] = {
+    mainConnection = createMainConnection()
+    mainStatement = mainConnection.createStatement()
+    mainResultSet = mainStatement.executeQuery(queryString).asInstanceOf[EXAResultSet]
+
+    val handle = mainResultSet.GetHandle()
```

Comment:
I still could not understand the logic behind this method, in this case only one value will be used across all the partitions.

Reply:
Hey @3cham,
Yes, it is a single value used across all partitions. Here, we could just provide it to the compute since it is a single integer; however, I put it into each partition. This might help if the value changes.

Reply:
Hi @morazow, then there is no difference between with and without handle? Do you have any ideas if the exa slave connections will be improved in the future?

Reply:
Hello @3cham ,

That is correct. I wanted to push still this pr since it is more clearer what is going on underhood.

Good question, there might be changes / updates but in later versions of Exasol. Additionally, I will try to change a little bit the architecture to early start the sub-connections. It is just an idea, currently I am not sure if it would help with the issue where we have less task from Spark side.